reqactor/src/actor.rs

```diff
@@ -132,8 +132,17 @@ impl Actor {
             return Ok(pool_status_opt.unwrap());
         }
 
-        // Mark the request as registered in the pool
-        let status = StatusWithContext::new(Status::Registered, start_time);
+        // Mark the request as registered in the pool, or update to work in progress if already registered
```

Comment:
better to have this in another PR on main branch separately, in case there is a bug then we need to revert whole boundless integration.